rust/gitxetcore/src/config/permission.rs

```diff
@@ -37,30 +54,125 @@ impl Permission {
         }
     }
 
-    pub fn check_path(&self, path: &Path) {
-        if self.is_elevated() && !path.exists() {
-            let message = format!("Warning: A xet command is running with elevated privileges. A xet metadata directory will be 
-created at {path:?} with elevated privileges. Future xet commands running with standard 
-privileges may not be able to access this folder, causing them to fail. If this is not desired, 
-please change the directory permissions accordingly.");
+    /// Check if the current process has R+W permission writing to 'path', if not can suggest an alternate
+    /// if 'suggest' is true by searching varies options in the below order:
+    ///
+    /// 1. paths in 'priority_suggest_list' from first to last.
+    /// 2. path_[$USER]
+    /// 3. $HOME/[suggest_prefix]_[random_number]
+    /// 4. $TMPDIR/[suggest_prefix]_[random_number]
+    /// 5. $PWD/[suggest_prefix]_[random_number]
+    /// The random numbers are generated from a CSPRNG and are unlikely to collide.
+    ///
+    /// If the current process is running with elevated privileges, warn if the checked path doesn't exist and will be created.
+    ///
+    /// Return Ok(path) if the provided path is good to write into or an alternate path is suggested;
+    /// Return Err(_) if lacking permision for the provided path and unable to suggest an alternate.
+    pub fn check_or_suggest_path(
```

Comment:
I'm trying to understand a use case for an alternate directory than `$HOME/.xet/`...  If it's a random path, or some other non-standard thing, isn't it essentially the same as using a directory created through the system temporary directory?   Is this just so that everything succeeds, even if that file isn't writable?

Reply:
The second search option `path_[$USER]` is to ensure a predictable path across git-xet calls. So there is a .xet folder already created with `sudo git-xet`, `git-xet` can still work and doesn't need to throw away cache, repos, or xet version check file. 

The rest three options are random trials to ensure that it always succeeds, with a slight preference to somewhere under `$HOME` because it's easier to find and (potentially) to be recovered from manually, because data transfer is billed. Same reasoning applies to constructing a predictable path name with `suggest_prefix` instead of from a pure random string.

Reply:
That's fair, but it could be surprising in a different way that if the user fixes their issue with ~/.xet, suddenly the cache and config values in any of these other directories would be suddenly ignored, which would effectively throw them away.